### PR TITLE
Publish working groups to publishing api

### PR DESCRIPTION
**Pull request by jamiecobbett**
We should probably wait for either https://github.com/alphagov/whitehall/pull/2478 or https://github.com/alphagov/publishing-api/pull/213 to be merged, so that we can be confident that the republishing step completes successfully, but I wanted to open this up for feedback now.

### CODE REVIEW

File: test/unit/presenters/publishing_api_presenters/working_group_test.rb [hunk 1]
```diff
@@ -25,12 +28,24 @@ class PublishingApiPresenters::WorkingGroupTest < ActiveSupport::TestCase
...
+    assert_valid_against_schema(presenter.content, 'working_group')
+  end
+
+  test "renders attachments in the body" do
+    group = create(:policy_group, :with_file_attachment, description: "#Heading\n\n!@1\n\n##Subheading")
+
+    presenter = PublishingApiPresenters::WorkingGroup.new(group)
+
+    body = Nokogiri::HTML.parse(presenter.content[:details][:body])
+    assert_not_nil body.at_css("section.attachment")
```
Comment (jamiecobbett) [inline test/unit/presenters/publishing_api_presenters/working_group_test.rb]: This is the best assertion I could come up with for demonstrating the attachment rendered, without doing a full string comparison. I'm open to alternatives.

Comment (boffbowsh) [inline test/unit/presenters/publishing_api_presenters/working_group_test.rb]: Maybe [pseudocode]: 
```ruby
assert_match %r{#{group.attachments.first.title}}, body.at_css("section.attachment")
```

**Comment by boffbowsh** [comment]
:+1: apart from maybe the comment

File: test/unit/presenters/publishing_api_presenters/working_group_test.rb (hunk 1 above)
Comment (jamiecobbett) [inline test/unit/presenters/publishing_api_presenters/working_group_test.rb]: Sounds reasonable - rebased to add it as a second assertion.

**Comment by boffbowsh** [comment]
:+1:

**Comment by jamiecobbett** [comment]
In the interests of broadcasting information, I've found a bug in the attachment rendering that prevents some working groups from being published.

**Comment by jamiecobbett** [comment]
OK, one attachment rendering bug fixed, another to go.

**Comment by jamiecobbett** [comment]
Here's the stack trace for the second attachment rendering bug:
```
Actor crashed!
ActionView::Template::Error: undefined method `safe_append=' for nil:NilClass
	/var/govuk/whitehall/app/views/documents/_inline_attachment.html.erb:2:in `_app_views_documents__inline_attachment_html_erb__215693695209900549_69887307925260'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/template.rb:145:in `block in render'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.2.5.2/lib/active_support/notifications.rb:166:in `instrument'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/template.rb:333:in `instrument'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/template.rb:143:in `render'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/renderer/partial_renderer.rb:339:in `render_partial'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/renderer/partial_renderer.rb:310:in `block in render'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/renderer/abstract_renderer.rb:39:in `block in instrument'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.2.5.2/lib/active_support/notifications.rb:164:in `block in instrument'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.2.5.2/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.2.5.2/lib/active_support/notifications.rb:164:in `instrument'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/renderer/abstract_renderer.rb:39:in `instrument'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/renderer/partial_renderer.rb:309:in `render'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/renderer/renderer.rb:51:in `render_partial'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/renderer/renderer.rb:25:in `render'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/actionview-4.2.5.2/lib/action_view/helpers/rendering_helper.rb:32:in `render'
	/var/govuk/whitehall/app/helpers/govspeak_helper.rb:235:in `block in govspeak_with_attachments_and_alt_format_information'
	/var/govuk/whitehall/app/helpers/govspeak_helper.rb:233:in `gsub'
	/var/govuk/whitehall/app/helpers/govspeak_helper.rb:233:in `govspeak_with_attachments_and_alt_format_information'
	/var/govuk/whitehall/app/helpers/govspeak_helper.rb:31:in `bare_govspeak_with_attachments_to_html'
	/var/govuk/whitehall/app/helpers/govspeak_helper.rb:21:in `govspeak_with_attachments_to_html'
	/var/govuk/whitehall/lib/whitehall/govspeak_renderer.rb:4:in `govspeak_with_attachments_to_html'
	/var/govuk/whitehall/app/presenters/publishing_api_presenters/working_group.rb:26:in `body'
	/var/govuk/whitehall/app/presenters/publishing_api_presenters/working_group.rb:20:in `details'
	/var/govuk/whitehall/app/presenters/publishing_api_presenters/item.rb:26:in `content'
	/var/govuk/whitehall/app/workers/publishing_api_worker.rb:37:in `save_draft'
	/var/govuk/whitehall/app/workers/publishing_api_worker.rb:32:in `send_item'
	/var/govuk/whitehall/app/workers/publishing_api_worker.rb:12:in `block in perform'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/i18n-0.7.0/lib/i18n.rb:257:in `with_locale'
	/var/govuk/whitehall/app/workers/publishing_api_worker.rb:10:in `perform'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/processor.rb:80:in `execute_job'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/processor.rb:56:in `block (2 levels) in process'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/chain.rb:127:in `block in invoke'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-statsd-0.1.5/lib/sidekiq/statsd/server_middleware.rb:44:in `block (2 levels) in call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/statsd-ruby-1.2.1/lib/statsd.rb:232:in `time'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-statsd-0.1.5/lib/sidekiq/statsd/server_middleware.rb:43:in `block in call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/statsd-ruby-1.2.1/lib/statsd.rb:67:in `easy'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/statsd-ruby-1.2.1/lib/statsd.rb:248:in `batch'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-statsd-0.1.5/lib/sidekiq/statsd/server_middleware.rb:38:in `call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/chain.rb:129:in `block in invoke'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/server/active_record.rb:6:in `call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/chain.rb:129:in `block in invoke'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/server/retry_jobs.rb:74:in `call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/chain.rb:129:in `block in invoke'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/server/logging.rb:11:in `block in call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/logging.rb:30:in `with_context'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/server/logging.rb:7:in `call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/chain.rb:129:in `block in invoke'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/chain.rb:132:in `call'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/middleware/chain.rb:132:in `invoke'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/processor.rb:51:in `block in process'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/processor.rb:104:in `stats'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/sidekiq-3.5.4/lib/sidekiq/processor.rb:50:in `process'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/call/async.rb:7:in `dispatch'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
	/usr/lib/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'"
```

**Comment by jamiecobbett** [comment]
OK, I've successfully republished all 630 Working Groups in my dev database without error, and the sync check was successful.

**Comment by gpeng** [comment]
Woohoo! 💥

**Comment by boffbowsh** [comment]
:+1:

**Pull request merged**